apis/refs/v1beta1/projectref.go

```diff
@@ -166,5 +166,13 @@ func ResolveProjectID(ctx context.Context, reader client.Reader, obj *unstructur
 		return projectID, nil
 	}
 
+	if selfLink, _, _ := unstructured.NestedString(obj.Object, "status", "selfLink"); selfLink != "" {
```

Comment:
Did you hit an actual case that doesnot have the project ID from the spec.projectID or the annotation?

Reply:
In the mock test, there is no annotation for project ID and this resource does not have a spec.projectID. I thought about adding the annotation in the yamls in mock tests, but I am not sure if I am supposed to change the create/update yamls in the mock test when adding the direct controller since it was working with TF.